frig-js/frig
# Initial setup of mocha/chai/jsdom testing infrastructure

This PR

* Removes previous test files
* Adds devDependencies `chai`, `mocha`, `enzyme`, `dirty-chai`, `jsdom`
* Adds `react-dom` to peerDependencies and devDependencies
* Fixes bug in `src/higher_order_components/higher_order_component.js`, was missing an arrow function
* Adds npm scripts `test` and `test:watch`
* Adds `test/test_helper.js` for setting up mocha with jsdom
* **Adds tests** for:
  * `<Form/>` (including `AbstractForm`)
  * `src/util.js`
  * `src/validations.js`

Refactorings:

* **BREAKING**: In `src/index.js`, do not `export default` an object, rather use only named exports with no default export. 
  * This means `Frig.defaultTheme()` will no longer work (consumers would have to `import { defaultTheme } from 'frig'`. 
  * It's probably worth considering special-casing just that one import, so we can retain the old syntax. 
  * *([Read this for justification, ^F for "Why do we need named exports?"](http://www.2ality.com/2014/09/es6-modules-final.html))*

* In `src/components/abstract_form.js`, remove useless `_data()` method

* In `src/higher_order_components/errors_normalizer.js`, remove dead code from`_errors()`

* Change `src/validations.js` from CommonJS to ES6 module

* Add names to all anonymous classes

* Remove all dead code from `src/util.js` (most of the file!)
